clients/drcachesim/tools/view.cpp

```diff
@@ -591,6 +591,11 @@ view_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
         return true;
     }
 
+    // XXX: We could potentially use instr_decode_cache_t here (i#7113) and avoid the
+    // repeated instr decoding logic. However, we want to preserve the legacy view
+    // tool output format which uses disassemble_to_buffer, and disassemble_to_buffer
+    // does the decoding on its own internally, while adding a bunch of non-trival
```

Comment:
You would switch to using instr_disassemble().

Reply:
Seems fine to update the view tool separately (this PR is already large), so long as we know it won't require interface changes to this library.  If it used instr_disassemle(), and the library sets ISA regdeps, will it all work out?

Reply:
To clarify: this would require some refactoring of the disassemble library to share the additional logic disassemble_to_buffer has today over instr_disassemble.

> so long as we know it won't require interface changes to this library.

I don't think it would. It's more about how view_t uses the provided decoded instr_t, and for instr_disassemble to provide the functionality offered by disassemble_to_buffer.

Reply:
decode_from_copy doesn't seem to set the raw bytes in the instr_t, without which adding the encoding to view_t output would be difficult. I guess decode_info_base_t::set_decode_info_derived can possibly re-encode the instr to get those bytes; another way would be that decode_cache_t also provide the decode_pc to decode_info_base_t::set_decode_info_derived (with a clear note that the address is valid only for that call).

Reply:
Or: before passing on the instr_t to decode_info_base_t, decode_cache_t could do instr_set_raw_bits to the decode_pc (better than `instr_allocate_raw_bits+instr_set_raw_bytes` as it avoids invalidating the operands etc).

Reply:
> Or: before passing on the instr_t to decode_info_base_t, decode_cache_t could do instr_set_raw_bits to the decode_pc (better than `instr_allocate_raw_bits+instr_set_raw_bytes` as it avoids invalidating the operands etc).

Ah, but the problem is that: decode_pc cannot be guaranteed to stay valid for embedded-encodings where it lives as part of the memref_t which doesn't stay around after the process_memref/parallel_shard_memref.